Automattic/pocket-casts-android
# Make trial logic generic for offers

## Description
- This PR does a little refactor for subscriptions with an offer. Currently we have only trial subscription and we want to support another type of offer
- This prepared for an update on the upgrade subscription page. In the next PR I will implement the UI, but for now I replaced the offer strings with `TODO`

## Screenshot
<img src="https://github.com/Automattic/pocket-casts-android/assets/42220351/66bd7215-8715-455a-bfcd-894502236e18" height="500">

## Testing
git apply [release.patch](https://github.com/Automattic/pocket-casts-android/files/13998087/release.patch)

1. Run the app and login in with a free account that never had a subscription
2. Go to `Podcasts` tab
3. Click on folders
**4. You should see the TODO text on the upgrade Button for Plus/Yearly plan** 


## Checklist
